### ISSUE TITLE
Labels with Nested Inputs: label-has-for

### ISSUE THREAD

**Issue by jacobp100** [evcohen/eslint-plugin-jsx-a11y#104]
When turned on, this rule incorrectly shows errors for the following code snippet.

```html
<label>
  <input type="checkbox" />
  Option
</label>
```

**Comment by ljharb** [comment]
The errors are correct. You need to *both* nest, and for-ID link, to get full a11y coverage of all assistive devices.

**Comment by jacobp100** [comment]
The spec states that nesting as demonstrated has the same effect as using the `for` attribute. What's the benefit of adding the attribute in this case?

**Comment by ljharb** [comment]
@jacobp100 Not every device follows the spec. Some 95% of devices work with nesting alone, a different 95% work with for-ID linking alone - the only way to cover 100% is to do both. Accessibility is about 100% coverage, so nothing short of that is "good enough".

**Comment by CyrilWork** [comment]
I don't know if it is the good place to ask, but where can I contribute code to make that 95% be 100% ?

**Comment by ljharb** [comment]
@CyrilWork nowhere, unless you have a time machine - the comparison would be asking how you could make IE 9 support flexbox, as an example.

**Comment by CyrilWork** [comment]
I will think about it
